### PR TITLE
Support passing HNSW config to QdrantDocumentStore

### PR DESCRIPTION
**Pull request by danielfleischer**
Including example in README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ document_store = QdrantDocumentStore(
     url="localhost",
     index="Document",
     embedding_dim=512,
+    hnsw_config={"m": 16, "ef_construct": 64},
```
Comment (kacperlukawski) [inline README.md:34]: Could you please make the parameter optional and pass `None` as the default value? I want to make sure the client defaults are always used.

File: src/qdrant_haystack/document_stores/qdrant.py [hunk 2]
```diff
@@ -48,6 +48,7 @@ def __init__(
         host: Optional[str] = None,
         index: str = "Document",
         embedding_dim: int = 768,
+        hnsw_config: dict = None,
```
Comment (kacperlukawski) [inline src/qdrant_haystack/document_stores/qdrant.py:51]: I'd also suggest typing it as `Optional`

**Review by kacperlukawski — CHANGES_REQUESTED** [review]
Looks fine - there is one thing I'd prefer to change

File: README.md (hunk 1 above)
Comment (danielfleischer) [reply README.md]: It was made optional, has a default `None` value and clarification in the README.

**Review by kacperlukawski — APPROVED** [review]

**Pull request merged**